### PR TITLE
MongoDB data fixtures

### PR DESCRIPTION
**Pull request by apohrebniak**
This pull request adds an ability to use Mongo fixtures with SpringRunner. **MongoFixture** annotation could be added to test classes and test methods

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@apohrebniak Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=12016)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-boot&pullRequestId=12016) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@apohrebniak Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=12016)!

**Comment by pivotal-issuemaster** [comment]
@apohrebniak Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=12016)!

File: spring-boot-project/spring-boot-test-autoconfigure/src/test/java/org/springframework/boot/test/autoconfigure/data/mongo/DataMonoTestFixtures.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringRunner.class)
+@DataMongoTest
+@UseMongoFixtures
+@MongoFixture(collection = "books", value = {
+		"org/springframework/boot/test/autoconfigure/data/mongo/books.fixture" })
+public class DataMonoTestFixtures {
```
Comment (Chr1st0ph) [inline spring-boot-project/spring-boot-test-autoconfigure/src/test/java/org/springframework/boot/test/autoconfigure/data/mongo/DataMonoTestFixtures.java]: should this be called DataMongoTestFixtures ?

Comment (apohrebniak) [reply spring-boot-project/spring-boot-test-autoconfigure/src/test/java/org/springframework/boot/test/autoconfigure/data/mongo/DataMonoTestFixtures.java]: changed this typo

**Comment by snicoll** [comment]
@apohrebniak thanks for the PR. The low-level support does not belong here but in Spring Data MongoDB so I suggest to check with the team to see if there's an interest for that feature. Should that be the case, we can revisit the integration with `@DataMongoTest` later.

**Pull request closed without merging**